pkg/component/worker/kubelet.go

```diff
@@ -131,6 +132,27 @@ func (k *Kubelet) Init(_ context.Context) error {
 	return nil
 }
 
+func getIPs(ctx context.Context, hostname string) (net.IP, net.IP, error) {
+	ipaddrs, err := net.DefaultResolver.LookupIPAddr(ctx, hostname)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ipv4, ipv6 net.IP
+	for _, addr := range ipaddrs {
+		if ipv4 == nil && addr.IP.To4() != nil {
+			ipv4 = addr.IP
+		} else if ipv6 == nil && addr.IP.To16() != nil && addr.IP.To4() == nil {
+			ipv6 = addr.IP
```

Comment:
```suggestion
			ipv6 = addr.IP.To16()
```

Reply:
Trying to wrap my head around what value this would bring?

In what situation would a valid IPv6 address not already be in a 16 byte representation?

Reply:
Yeah, this converts a v4 IP to a v6 IP, but as an earlier branch already prevents v4 IPs to reach this point, it's a no-op, indeed.

Thinking if lines 143 and following could better be written as

```go
		if v4 := addr.IP.To4(); v4 != nil {
			if ipv4 == nil {
				ipv4 = v4
			}
		} else if ipv6 == nil {
			if v6 := addr.IP.To16(); v6 != nil {
				ipv6 = v6
			}
		}
```